<?xml version="1.0" encoding="UTF-8"?><record_update table="sysauto_script">
    <sysauto_script action="INSERT_OR_UPDATE">
        <active>true</active>
        <business_calendar/>
        <condition/>
        <conditional>false</conditional>
        <entered_time>1970-01-01 00:00:00</entered_time>
        <max_drift/>
        <name>Pull Cloud Assets data</name>
        <offset/>
        <offset_type>0</offset_type>
        <run_as display_value="System Administrator">6816f79cc0a8016401c5a33be04be441</run_as>
        <run_as_tz/>
        <run_dayofmonth>1</run_dayofmonth>
        <run_dayofweek>1</run_dayofweek>
        <run_period/>
        <run_start>2024-08-21 10:16:11</run_start>
        <run_time>1970-01-01 08:00:00</run_time>
        <run_type>on_demand</run_type>
        <script><![CDATA[try {
  var provider = new sn_cc.ConnectionInfoProvider();
  var connectionInfo = provider.getConnectionInfo("a6b16dd31bc8969028b599f91d4bcb8c");
  gs.info("getting connection info");
  if (connectionInfo != null) {
	var url = connectionInfo.getAttribute("host");
	var customerId = connectionInfo.getCredentialAttribute("user_name");
	var iss = connectionInfo.getCredentialAttribute("privacy_key");
	var secret = connectionInfo.getCredentialAttribute("password");
  } else {
	gs.info("ConnectionInfo is null. Script execution halted.");
	throw new Error("ConnectionInfo is null. Script execution halted.");
  }

  var comp = new UptycsAPI();
  comp.initialize(customerId, url, iss, secret);
  gs.info('Fetching S3 buckets data');
  comp.ingestS3BucketRecords();
  gs.info('Fetching ECS Clusters data');
  comp.ingestECSClusterRecords();
  gs.info('Fetching Lambda fucntions data');
  comp.ingestLambdaFunctionRecords();
  gs.info('Fetching AWS IAM data');
  comp.ingestAWSIAMRecords();
  gs.info('Fetching RDS clusters data');
  comp.ingestRDSClustersRecords();
  gs.info('Fetching Azure sql data');
  comp.ingestAzureSqlRecords();

  gs.warn('End Uptycs Cloud Inventory import');
	
} catch(ex) {
  gs.warn("Scheduled Script Execution: 'Pull Uptycs cloud Inventory' failed with error message: " + ex.message);
}]]></script>
        <sys_class_name>sysauto_script</sys_class_name>
        <sys_created_by>admin</sys_created_by>
        <sys_created_on>2024-08-21 10:17:09</sys_created_on>
        <sys_id>fa04e5db1bc8969028b599f91d4bcb88</sys_id>
        <sys_mod_count>13</sys_mod_count>
        <sys_name>Pull Cloud Assets data</sys_name>
        <sys_package display_value="Uptycs" source="x_297025_uptycs_co">31b4358e2f2630105113fe1df699b66d</sys_package>
        <sys_policy/>
        <sys_scope display_value="Uptycs">31b4358e2f2630105113fe1df699b66d</sys_scope>
        <sys_update_name>sysauto_script_fa04e5db1bc8969028b599f91d4bcb88</sys_update_name>
        <sys_updated_by>admin</sys_updated_by>
        <sys_updated_on>2024-08-22 11:54:12</sys_updated_on>
        <time_zone/>
        <upgrade_safe>false</upgrade_safe>
    </sysauto_script>
</record_update>
